ssl/ssl_sess.c

```diff
@@ -244,6 +245,15 @@ SSL_SESSION *ssl_session_dup(SSL_SESSION *src, int ticket)
     }
 #endif
 
+    if (src->ticket_appdata != NULL) {
+        dest->ticket_appdata =
+            OPENSSL_memdup(src->ticket_appdata, src->ticket_appdata_len);
+        if (dest->ticket_appdata == NULL)
+            goto err;
+    } else {
+        dest->ticket_appdata = 0;
```

Comment:
Don't we also need to set the `ticket_appdata_len` in both of these cases? I'm wondering why we don't zalloc dest to start with, which would at least simplify this.

Also, you already explicitly initialised `dest->ticket_appdata` to `NULL` earlier in this function, so this last line looks redundant.

Reply:
Yes, and it's being set to 0 rather than NULL.

We don't zalloc() because of the shallow copy `memcpy(dest, src, sizeof(*dest));` at line 113. The destination pointers are reset, and then a deep copy is made of each of them.

The copy of 'ticket_appdata_len` is not necessary because of the shallow copy.